ocamltest/ocamltest_stdlib.ml

```diff
@@ -84,22 +84,36 @@ end
 module Sys = struct
   include Sys
 
-  let run_system_command prog args =
-    let command = Filename.quote_command prog args in
-    match Sys.command command with
-    | 0 -> ()
-    | _ as exitcode ->
-      Printf.eprintf "System command %s failed with status %d\n%!"
-        command exitcode;
-      exit 3
-
-  let mkdir dir =
-    if not (Sys.file_exists dir) then
-      run_system_command "mkdir" [dir]
+  let erase_file path =
+    try Sys.remove path
+    with Sys_error _ when Sys.win32 && Ocamltest_config.libunix <> None ->
+      (* Deal with read-only attribute on Windows. Ignore any error from chmod
+         so that the message always come from Sys.remove *)
+      let () = try Unix.chmod path 0o666 with Sys_error _ -> () in
+      Sys.remove path
+
+  let rm_rf path =
+    let rec erase path =
+      if Sys.is_directory path
+      then Array.iter (fun entry -> erase (Filename.concat path entry))
+                      (Sys.readdir path)
+      else erase_file path
+    in
+      try if Sys.file_exists path then erase path
+      with Sys_error _ ->
+        raise (Sys_error (Printf.sprintf "rm_rf failed on %S" path))
```

Comment:
I'm sorry for being difficult, but this error handling is not very useful.  When `erase_file` raises a `Sys_error` exception, the argument to the exception gives both the path name that fails and the reason for failure (e.g. "permission denied" or "readonly file system" or "no such file" or what not).  The reason for failure is good to know.  Here you're hiding it.

Plus, I don't like error messages that mention OCaml function names, because users don't care about which OCaml function failed, they care about what went wrong.

Reply:
Ah, I forgot that the error message is included and not just the path. That should most certainly be fixed, I agree.

The reason behind the function name was because there's no context in the `Sys_error`:
```
# open_out "system32";;
Exception: Sys_error "system32: Permission denied".
# Sys.remove "system32";;
Exception: Sys_error "system32: Permission denied".
```
but there is in `Unix_error` (and in `Unix.handle_error`).

Could I solve this by changing it to:

```ocaml
      with Sys_error msg ->
        raise (Sys_error (Printf.sprintf "Failed to remove %S (%s)" path msg))
```
?

Reply:
```
with Sys_error msg ->
        raise (Sys_error (Printf.sprintf "Failed to remove %S (%s)" path msg))
```
That would be better, yes.

Reply:
Both fixed.